olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -555,81 +564,38 @@ client::CancellationToken AuthenticationClientImpl::HandleUserRequest(
-  request.WithHeader(http::kUserAgentHeader, http::kOlpSdkUserAgent);
-  request.WithSettings(std::move(network_settings));
+  auto signup_task = [=](client::CancellationContext context) -> ResponseType {
+    if (!settings_.network_request_handler) {
+      return client::ApiError::NetworkConnection(
+          "Cannot sign up while offline");
+    }
 
-  std::shared_ptr<std::stringstream> payload =
-      std::make_shared<std::stringstream>();
-  request.WithBody(GenerateSignUpBody(properties));
-  auto send_outcome = settings_.network_request_handler->Send(
-      request, payload,
-      [callback, payload,
-       credentials](const http::NetworkResponse& network_response) {
-        auto response_status = network_response.GetStatus();
-        auto error_msg = network_response.GetError();
+    client::OlpClient client = CreateOlpClient(settings_, {}, false);
 
-        if (response_status < 0) {
-          // Network error response
-          AuthenticationError error(response_status, error_msg);
-          callback(error);
-          return;
-        }
+    const auto url = settings_.token_endpoint_url + kUserEndpoint;
+    auto auth_header = GenerateAuthorizationHeader(
+        credentials, url, std::time(nullptr), GenerateUid());
 
-        auto document = std::make_shared<rapidjson::Document>();
-        rapidjson::IStreamWrapper stream(*payload.get());
-        document->ParseStream(stream);
+    client::OlpClient::ParametersType headers = {
+        {http::kAuthorizationHeader, std::move(auth_header)}};
 
-        std::shared_ptr<SignUpResultImpl> resp_impl =
-            std::make_shared<SignUpResultImpl>(response_status, error_msg,
-                                               document);
-        SignUpResult response(resp_impl);
-        callback(response);
-      });
+    auto signup_response = client.CallApi(kUserEndpoint, "POST", {}, headers,
+                                          {}, GenerateSignUpBody(properties),
+                                          kApplicationJson, std::move(context));
 
-  if (!send_outcome.IsSuccessful()) {
-    thread::ExecuteOrSchedule(settings_.task_scheduler, [send_outcome,
-                                                         callback] {
-      std::string error_message =
-          ErrorCodeToString(send_outcome.GetErrorCode());
-      AuthenticationError result({static_cast<int>(send_outcome.GetErrorCode()),
-                                  std::move(error_message)});
-      callback(result);
-    });
-    return client::CancellationToken();
-  }
+    const auto status = signup_response.status;
+    if (status < 0) {
+      return client::ApiError(signup_response.status,
+                              signup_response.response.str());
+    }
 
-  auto request_id = send_outcome.GetRequestId();
-  return client::CancellationToken([weak_network, request_id]() {
-    auto network = weak_network.lock();
+    return ParseSignUpResponse(status, std::move(signup_response.response));
```

Comment:
Well, if you have a look at HttpResponse.response it says you should not use it directly as it is deprecated.
Please use GetResponse(), and if this is not enough for you, extend it with another GetResponse() method which will grant you access to the underlying stream.

Reply:
done